src/YesSql.Core/Services/DefaultQuery.cs

```diff
@@ -1302,6 +1313,17 @@ IQuery<T> IQuery<T>.All(params Func<IQuery<T>, IQuery<T>>[] predicates)
                 return query;
             }
 
+            ValueTask<IQuery<T>> IQuery<T>.AllAsync(params Func<IQuery<T>, ValueTask<IQuery<T>>>[] predicates)
+            {
+                // Scope the currentPredicate so multiple calls will not act on the new predicate.
+                var currentPredicate = _query._queryState._currentPredicate;
+                var query = ComposeQueryAsync(predicates, new AndNode());
+                // Return the currentPredicate to it's previous value, so another method call will act on the previous predicate.
+                _query._queryState._currentPredicate = currentPredicate;
```

Comment:
@deanmarcussen 

Just after a quick look so not sure, If `ComposeQueryAsync()` is not awaited and mutates `_currentPredicate`, it may mutate it after it is retrieved line 1322

Reply:
You might be right there. I will check tomorrow cheers :)